ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/MainActivity.java

```diff
@@ -12,34 +15,38 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        Button buttons[] = new Button[4];
-        buttons[0] = (Button) findViewById(R.id.btn_activity1);
-        buttons[1] = (Button) findViewById(R.id.btn_activity2);
-        buttons[2] = (Button) findViewById(R.id.btn_activity3);
-        buttons[3] = (Button) findViewById(R.id.btn_activity4);
+        final Button button = (Button) findViewById(R.id.btn_activity1);
+        button.setOnClickListener(this);
+        button.setEnabled(false);
 
-        for (Button b: buttons) {
-            b.setOnClickListener(this);
-        }
+        EditText edit = (EditText) findViewById(R.id.edit_txt);
+        assert edit != null;
+        edit.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                if (s.length() == 0) {
+                    button.setEnabled(false);
+                    return;
+                }
+
+                button.setEnabled(true);
```

Comment:
this is ok, can also be written as `button.setEnabled(s.length() != 0);` i think